packages/require/lib/compiler.ts

```diff
@@ -0,0 +1,94 @@
+import path from "path";
+import TruffleError from "@truffle/error";
+
+import type { CreateOptions, Service } from "ts-node";
+import TruffleConfig from "@truffle/config";
+import { readFileSync, existsSync } from "fs";
+
+import originalRequire from "original-require";
+
+import vm from "vm";
+
+/**
+ * @returns compiler function from user's installation of `ts-node` if script is
+ * a ts or tsx file, otherwise `undefined`.
+ */
+export function compile(conf: TruffleConfig, scriptPath: string): string {
+  const source = readFileSync(scriptPath, { encoding: "utf-8" });
+
+  // we only compile TS files, so just return the source unless we are dealing
+  // with one of those
+  if (![".ts", ".tsx", ".cts"].includes(path.extname(scriptPath))) {
+    return source;
+  }
```

Comment:
In theory we could pass `js` files through this, especially in the case where the user has `allowJs: true` in their `tsconfig.json`.

I decided against that however because 1) I'd need to add logic to detect this condition, and 2) I figured it'd be best to avoid performance degradation for existing projects.

Reply:
Good question. I purposefully excluded `.mts` files for now, as I didn't want the added complexity of dealing with EJS modules. I suppose I can imagine a world in which users will want EJS support, but for now I'm not sure what'd be driving that desire in migration script code other than a preference to use what's new.

Reply:
resolving w/o modification, but please comment if you feel that `mts` should be in the list.